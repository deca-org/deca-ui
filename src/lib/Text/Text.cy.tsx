import { Test } from '../Utils';
import _cyp from '../../../cypress';
import React from 'react';
import Text, { TextLineHeight, TextWeight } from './Text';

const allWeights = [
  'hairline',
  'thin',
  'light',
  'normal',
  'medium',
  'semibold',
  'bold',
  'extrabold',
  'black',
];

const allLineHeights = ['n', '0', '1', '2', '3', '4', '5', '6'];

describe('components/Text', () => {
  describe('base', () => {
    it('no margin', () => {
      cy.mount(<Text as="h1">Hello World</Text>);
      cy.get('h1').should('have.css', 'margin', Test.space('n'));
    });
    it('text color', () => {
      cy.mount(<Text as="h1">Hello World</Text>);
      cy.get('h1').should('have.css', 'color', Test.color('black'));
    });
  });
  describe('as', () => {
    it('h1', () => {
      cy.mount(<Text as="h1">Hello World</Text>);
      cy.get('h1').should('have.css', 'font-size', Test.fontSize('h1'));
      cy.get('h1').should('have.css', 'line-height', Test.lineHeight('6'));
    });
    it('h2', () => {
      cy.mount(<Text as="h2">Hello World</Text>);
      cy.get('h2').should('have.css', 'font-size', Test.fontSize('h2'));
      cy.get('h2').should('have.css', 'line-height', Test.lineHeight('5'));
    });
    it('h3', () => {
      cy.mount(<Text as="h3">Hello World</Text>);
      cy.get('h3').should('have.css', 'font-size', Test.fontSize('h3'));
      cy.get('h3').should('have.css', 'line-height', Test.lineHeight('4'));
    });
    it('h4', () => {
      cy.mount(<Text as="h4">Hello World</Text>);
      cy.get('h4').should('have.css', 'font-size', Test.fontSize('h4'));
      cy.get('h4').should('have.css', 'line-height', Test.lineHeight('3'));
    });
    it('h5', () => {
      cy.mount(<Text as="h5">Hello World</Text>);
      cy.get('h5').should('have.css', 'font-size', Test.fontSize('h5'));
      cy.get('h5').should('have.css', 'line-height', Test.lineHeight('2'));
    });
    it('h6', () => {
      cy.mount(<Text as="h6">Hello World</Text>);
      cy.get('h6').should('have.css', 'font-size', Test.fontSize('h6'));
      cy.get('h6').should('have.css', 'line-height', Test.lineHeight('2'));
    });
    it('p', () => {
      cy.mount(<Text as="p">Hello World</Text>);
      cy.get('p').should('have.css', 'font-size', Test.fontSize('body'));
      cy.get('p').should('have.css', 'line-height', Test.lineHeight('1'));
    });
    it('span', () => {
      cy.mount(<Text as="span">Hello World</Text>);
      cy.get('span').should('have.css', 'font-size', Test.fontSize('body'));
      cy.get('span').should('have.css', 'line-height', Test.lineHeight('1'));
    });
    it('blockquote', () => {
      cy.mount(<Text as="blockquote">Hello World</Text>);
      cy.get('blockquote').should(
        'have.css',
        'font-size',
        Test.fontSize('body')
      );
      cy.get('blockquote').should(
        'have.css',
        'line-height',
        Test.lineHeight('1')
      );
      cy.get('blockquote').should('have.css', 'margin-left', Test.space('4'));
    });
    it('b', () => {
      cy.mount(<Text as="b">Hello World</Text>);
      cy.get('b').should('have.css', 'font-size', Test.fontSize('body'));
      cy.get('b').should('have.css', 'line-height', Test.lineHeight('1'));
    });
    it('small', () => {
      cy.mount(<Text as="small">Hello World</Text>);
      cy.get('small').should('have.css', 'font-size', Test.fontSize('caption'));
      cy.get('small').should('have.css', 'line-height', Test.lineHeight('0'));
    });
    it('del', () => {
      cy.mount(<Text as="del">Hello World</Text>);
      cy.get('del').should('have.css', 'font-size', Test.fontSize('body'));
      cy.get('del').should('have.css', 'line-height', Test.lineHeight('1'));
    });
    it('i', () => {
      cy.mount(<Text as="i">Hello World</Text>);
      cy.get('i').should('have.css', 'font-size', Test.fontSize('body'));
      cy.get('i').should('have.css', 'line-height', Test.lineHeight('1'));
    });
    it('em', () => {
      cy.mount(<Text as="em">Hello World</Text>);
      cy.get('em').should('have.css', 'font-size', Test.fontSize('body'));
      cy.get('em').should('have.css', 'line-height', Test.lineHeight('1'));
    });
  });
  describe('size', () => {
    it('h1 with size p', () => {
      cy.mount(
        <Text as="h1" size="body" weight="normal">
          Hello World
        </Text>
      );
      cy.get('h1').should('have.css', 'font-size', Test.fontSize('body'));
      cy.get('h1').should('have.css', 'line-height', Test.lineHeight('1'));
      cy.get('h1').should('have.css', 'font-weight', Test.fontWeight('normal'));
    });
    it('weight overrides size weight', () => {
      cy.mount(
        <Text as="h1" size="body" weight="hairline">
          Hello World
        </Text>
      );
      cy.get('h1').should('have.css', 'font-size', Test.fontSize('body'));
      cy.get('h1').should('have.css', 'line-height', Test.lineHeight('1'));
      cy.get('h1').should(
        'have.css',
        'font-weight',
        Test.fontWeight('hairline')
      );
    });
    it('lineHeight overrides size lineHeight', () => {
      cy.mount(
        <Text as="h1" size="body" lineHeight="4">
          Hello World
        </Text>
      );
      cy.get('h1').should('have.css', 'font-size', Test.fontSize('body'));
      cy.get('h1').should('have.css', 'line-height', Test.lineHeight('4'));
    });
  });
  describe('lineHeight', () => {
    allLineHeights.map((lineHeight) =>
      it(lineHeight, () => {
        cy.mount(
          <Text as="p" lineHeight={lineHeight as TextLineHeight}>
            Hello World
          </Text>
        );
        cy.get('p').should(
          'have.css',
          'line-height',
          Test.lineHeight(lineHeight)
        );
      })
    );
  });
  describe('font-weights', () => {
    allWeights.map((weight) =>
      it(weight, () => {
        cy.mount(
          <Text as="p" weight={weight as TextWeight}>
            Hello World
          </Text>
        );
        cy.get('p').should('have.css', 'font-weight', Test.fontWeight(weight));
      })
    );
  });
  it('center', () => {
    cy.mount(
      <Text as="h1" center>
        Hello World
      </Text>
    );
    cy.get('h1').should('have.css', 'text-align', 'center');
  });
  describe('dark mode', () => {
    it('text color', () => {
      cy.darkMount(<Text as="h1">Hello World</Text>);
      cy.get('h1').should('have.css', 'color', Test.color('white'));
    });
  });
});
