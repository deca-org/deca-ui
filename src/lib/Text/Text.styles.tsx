import { styled } from '@lib/Theme/stitches.config';

const StyledText = styled('p', {
  m: '$n',
  variants: {
    weight: {
      hairline: {
        fontWeight: '$hairline',
      },
      thin: {
        fontWeight: '$thin',
      },
      light: {
        fontWeight: '$light',
      },
      normal: {
        fontWeight: '$normal',
      },
      medium: {
        fontWeight: '$medium',
      },
      semibold: {
        fontWeight: '$semibold',
      },
      bold: {
        fontWeight: '$bold',
      },
      extrabold: {
        fontWeight: '$extrabold',
      },
      black: {
        fontWeight: '$black',
      },
    },
    as: {
      h1: {
        fontSize: '$h1',
        lineHeight: '$6',
      },
      h2: {
        fontSize: '$h2',
        lineHeight: '$5',
      },
      h3: {
        fontSize: '$h3',
        lineHeight: '$4',
      },
      h4: {
        fontSize: '$h4',
        lineHeight: '$3',
      },
      h5: {
        fontSize: '$h5',
        lineHeight: '$2',
      },
      h6: {
        fontSize: '$h6',
        lineHeight: '$2',
      },
      p: {
        fontSize: '$body',
        lineHeight: '$1',
      },
      span: {
        fontSize: '$body',
        lineHeight: '$1',
      },
      blockquote: {
        fontSize: '$body',
        lineHeight: '$1',
        ml: '$4',
      },
      b: {
        fontSize: '$body',
        lineHeight: '$1',
      },
      small: {
        fontSize: '$caption',
        lineHeight: '$0',
      },
      del: {
        fontSize: '$body',
        lineHeight: '$1',
      },
      i: {
        fontSize: '$body',
        lineHeight: '$1',
      },
      em: {
        fontSize: '$body',
        lineHeight: '$1',
      },
    },
    size: {
      h1: {
        fontSize: '$h1',
        lineHeight: '$6',
      },
      h2: {
        fontSize: '$h2',
        lineHeight: '$5',
      },
      h3: {
        fontSize: '$h3',
        lineHeight: '$4',
      },
      h4: {
        fontSize: '$h4',
        lineHeight: '$3',
      },
      h5: {
        fontSize: '$h5',
        lineHeight: '$2',
      },
      h6: {
        fontSize: '$h6',
        lineHeight: '$2',
      },
      body: {
        fontSize: '$body',
        lineHeight: '$1',
      },
      bodySm: {
        fontSize: '$bodySm',
        lineHeight: '$1',
      },
      caption: {
        fontSize: '$caption',
        lineHeight: '$1',
      },
      footnote: {
        fontSize: '$footnote',
        lineHeight: '$0',
      },
    },
    lineHeight: {
      n: {
        lineHeight: '$n',
      },
      0: {
        lineHeight: '$0',
      },
      1: {
        lineHeight: '$1',
      },
      2: {
        lineHeight: '$2',
      },
      3: {
        lineHeight: '$3',
      },
      4: {
        lineHeight: '$4',
      },
      5: {
        lineHeight: '$5',
      },
      6: {
        lineHeight: '$6',
      },
    },
    center: {
      true: {
        textAlign: 'center',
      },
      false: {},
    },
    isDark: {
      true: {
        color: '$text',
      },
      false: {},
    },
    mono: {
      true: {
        fontFamily: '$mono',
      },
      false: {
        fontFamily: '$normal',
      },
    },
  },
});

export default StyledText;
