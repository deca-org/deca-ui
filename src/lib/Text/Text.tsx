import { ThemeContext } from '@lib/Theme';
import { CSS } from '@lib/Theme/stitches.config';
import {
  PolymorphicRef,
  PolymorphicComponentPropWithRef,
  AnyObject,
  __DEV__,
} from '@lib/Utils';
import React from 'react';

import StyledText from './Text.styles';

export type TextAs =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'h5'
  | 'h6'
  | 'p'
  | 'span'
  | 'blockquote'
  | 'b'
  | 'small'
  | 'del'
  | 'i'
  | 'em';

export type TextWeight =
  | 'hairline'
  | 'thin'
  | 'light'
  | 'normal'
  | 'medium'
  | 'semibold'
  | 'bold'
  | 'extrabold'
  | 'black';

export type TextSize =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'h5'
  | 'h6'
  | 'body'
  | 'bodySm'
  | 'caption'
  | 'footnote'
  | AnyObject;

export type TextLineHeight = 'n' | '0' | '1' | '2' | '3' | '4' | '5' | '6';

/**
 * The Text component is the used to render text and paragraphs within an interface
 */
interface Props {
  /**
   * Changes which tag component outputs.
   */
  as?: TextAs;
  /**
   * Override default CSS style.
   */
  css?: CSS;
  /**
   * The content of the component.
   */
  children?: React.ReactNode | undefined;
  /**
   * Font weight of the text.
   */
  weight?: TextWeight;
  /**
   * Custom size for the text.
   */
  size?: TextSize;
  /**
   * Center text.
   */
  center?: boolean;
  /**
   * Custom line height for the text.
   */
  lineHeight?: TextLineHeight;
  /**
   * Should text be mono font-family.
   * @default false;
   */
  mono?: boolean;
}

export type TextProps<T extends React.ElementType> =
  PolymorphicComponentPropWithRef<T, Props>;

export type TextComponent = (<C extends React.ElementType = 'p'>(
  props: TextProps<C>
) => React.ReactElement | null) & { displayName?: string };

const Text: TextComponent = React.forwardRef(
  <T extends React.ElementType = 'p'>(
    {
      as,
      css,
      children,
      weight,
      size,
      lineHeight,
      center,
      mono = false,
      ...textProps
    }: TextProps<T>,
    ref?: PolymorphicRef<T>
  ) => {
    const preClass = 'decaText';

    const { dark } = React.useContext(ThemeContext);

    return (
      <StyledText
        as={as}
        css={css}
        className={`${preClass}-root`}
        ref={ref}
        weight={weight}
        center={center}
        size={size}
        isDark={dark}
        lineHeight={lineHeight}
        mono={mono}
        {...textProps}
      >
        {children}
      </StyledText>
    );
  }
);

if (__DEV__) {
  Text.displayName = 'DecaUI.Text';
}

export default Text;
