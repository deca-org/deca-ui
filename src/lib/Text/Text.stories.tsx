import { DecaUIProvider } from '@lib/Theme';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import Text from './Text';

export default {
  title: 'Text',
  component: Text,
} as ComponentMeta<typeof Text>;

const Template: ComponentStory<typeof Text> = (args) => <Text {...args} />;

const showcaseText = 'Almost before we knew it, we had left the ground.';

export const Header = Template.bind({});

Header.args = {
  as: 'h1',
  children: showcaseText,
  size: 'h1',
  mono: false,
};

export const Typescale = () => {
  return (
    <>
      <Text as="h1">{showcaseText}</Text>
      <Text as="h2">{showcaseText}</Text>
      <Text as="h3">{showcaseText}</Text>
      <Text as="h4">{showcaseText}</Text>
      <Text as="h5">{showcaseText}</Text>
      <Text as="h6">{showcaseText}</Text>
      <Text as="p" size="body">
        {showcaseText}
      </Text>
      <Text as="p">{showcaseText}</Text>
      <Text as="small">{showcaseText}</Text>
    </>
  );
};

export const WithTheme = Template.bind({});

WithTheme.args = { ...Header.args };
WithTheme.decorators = [
  (Story) => (
    <DecaUIProvider
      theme={{
        fonts: {
          normal: 'Times New Roman',
        },
      }}
    >
      <Story />
    </DecaUIProvider>
  ),
];

export const DarkMode = Template.bind({});

DarkMode.args = { ...Header.args };
DarkMode.parameters = { backgrounds: { default: 'dark' } };

DarkMode.decorators = [
  (Story) => (
    <DecaUIProvider mode="dark">
      <Story />
    </DecaUIProvider>
  ),
];
