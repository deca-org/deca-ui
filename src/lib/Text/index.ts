import Text from './Text';
export * from './Text';
export { Text };
export default Text;
