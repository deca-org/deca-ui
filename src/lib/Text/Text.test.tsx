import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';

import Text from './Text';

describe('components/Text', () => {
  it('renders content', () => {
    const { getByText } = render(<Text>Children</Text>);
    expect(getByText('Children')).toBeInTheDocument();
  });
  it('matches snapshot', () => {
    const { asFragment } = render(<Text></Text>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders all variants', () => {
    const { asFragment } = render(
      <>
        <Text as="h1">Children</Text>
        <Text as="h2">Children</Text>
        <Text as="h3">Children</Text>
        <Text as="h4">Children</Text>
        <Text as="h5">Children</Text>
        <Text as="h6">Children</Text>
        <Text as="span">Children</Text>
        <Text as="p">Children</Text>
        <Text as="blockquote">Children</Text>
        <Text as="b">Children</Text>
        <Text as="small">Children</Text>
        <Text as="del">Children</Text>
        <Text as="i">Children</Text>
        <Text as="em">Children</Text>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders all sizes', () => {
    const { asFragment } = render(
      <>
        <Text size="h1">Children</Text>
        <Text size="h2">Children</Text>
        <Text size="h3">Children</Text>
        <Text size="h4">Children</Text>
        <Text size="h5">Children</Text>
        <Text size="h6">Children</Text>
        <Text size="body">Children</Text>
        <Text size="bodySm">Children</Text>
        <Text size="caption">Children</Text>
        <Text size="footnote">Children</Text>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders all lineHeights', () => {
    const { asFragment } = render(
      <>
        <Text lineHeight="n">Children</Text>
        <Text lineHeight="0">Children</Text>
        <Text lineHeight="1">Children</Text>
        <Text lineHeight="2">Children</Text>
        <Text lineHeight="3">Children</Text>
        <Text lineHeight="4">Children</Text>
        <Text lineHeight="5">Children</Text>
        <Text lineHeight="6">Children</Text>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders sizes with variants', () => {
    const { asFragment } = render(
      <Text as="h1" size="h2">
        Children
      </Text>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
