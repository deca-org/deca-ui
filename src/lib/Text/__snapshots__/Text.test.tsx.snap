// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Text matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="c-hhwXGU c-hhwXGU-dTHfCl-mono-false decaText-root"
  />
</DocumentFragment>
`;

exports[`components/Text renders all lineHeights 1`] = `
<DocumentFragment>
  <p
    class="c-hhwXGU c-hhwXGU-hVSCMQ-lineHeight-n c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-kENwji-lineHeight-0 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-kbxdZX-lineHeight-1 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-bwqZvk-lineHeight-2 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-iwoKbl-lineHeight-3 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-guLIGy-lineHeight-4 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-fRvqxn-lineHeight-5 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-iBKVrw-lineHeight-6 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
</DocumentFragment>
`;

exports[`components/Text renders all sizes 1`] = `
<DocumentFragment>
  <p
    class="c-hhwXGU c-hhwXGU-cFWcZl-size-h1 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-kKUNrJ-size-h2 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-hlOZHJ-size-h3 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-erFqON-size-h4 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-cHMTHd-size-h5 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-cpZldy-size-h6 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-jQkWJT-size-body c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-RtCVl-size-bodySm c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-eqmpvd-size-caption c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <p
    class="c-hhwXGU c-hhwXGU-ejpLhE-size-footnote c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
</DocumentFragment>
`;

exports[`components/Text renders all variants 1`] = `
<DocumentFragment>
  <h1
    class="c-hhwXGU c-hhwXGU-cFWcZl-as-h1 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </h1>
  <h2
    class="c-hhwXGU c-hhwXGU-kKUNrJ-as-h2 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </h2>
  <h3
    class="c-hhwXGU c-hhwXGU-hlOZHJ-as-h3 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </h3>
  <h4
    class="c-hhwXGU c-hhwXGU-erFqON-as-h4 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </h4>
  <h5
    class="c-hhwXGU c-hhwXGU-cHMTHd-as-h5 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </h5>
  <h6
    class="c-hhwXGU c-hhwXGU-cpZldy-as-h6 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </h6>
  <span
    class="c-hhwXGU c-hhwXGU-jQkWJT-as-span c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </span>
  <p
    class="c-hhwXGU c-hhwXGU-jQkWJT-as-p c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </p>
  <blockquote
    class="c-hhwXGU c-hhwXGU-hfgHhY-as-blockquote c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </blockquote>
  <b
    class="c-hhwXGU c-hhwXGU-jQkWJT-as-b c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </b>
  <small
    class="c-hhwXGU c-hhwXGU-IpkNQ-as-small c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </small>
  <del
    class="c-hhwXGU c-hhwXGU-jQkWJT-as-del c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </del>
  <i
    class="c-hhwXGU c-hhwXGU-jQkWJT-as-i c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </i>
  <em
    class="c-hhwXGU c-hhwXGU-jQkWJT-as-em c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </em>
</DocumentFragment>
`;

exports[`components/Text renders sizes with variants 1`] = `
<DocumentFragment>
  <h1
    class="c-hhwXGU c-hhwXGU-cFWcZl-as-h1 c-hhwXGU-kKUNrJ-size-h2 c-hhwXGU-dTHfCl-mono-false decaText-root"
  >
    Children
  </h1>
</DocumentFragment>
`;
